### ISSUE TITLE
Knxd update fails with:  exit status 2

### ISSUE THREAD

**Issue by gjniewenhuijse** [knxd/knxd#206]
Sorry i'm a noob in Pi world.
Pi info: ARMv6-compatible processor rev 7 (v6l) / Linux version 4.1.19+ 

I have a running Pi with Domoticz image and knxd installed. Works perfect: version 0.10.0

I like to update to the latest version and did the following commands:
cd /home/pi
rm knxd*.deb
cd knxd
git pull
sudo dpkg-buildpackage -b -uc

The last command ends with:
configure: exit 1
dh_auto_configure: ./configure --build=arm-linux-gnueabihf --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/share/man --infodir=${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --libdir=${prefix}/lib/arm-linux-gnueabihf --libexecdir=${prefix}/lib/arm-linux-gnueabihf --disable-maintainer-mode --disable-dependency-tracking --enable-usb --enable-groupcache --enable-ft12 --enable-ncn5120 --enable-tpuarts --enable-tpuarts-tcp --enable-dummy --libexecdir=/usr/lib --enable-dependency-tracking --enable-silent-rules CFLAGS=-g -O0 CXXFLAGS=-g -O0 returned exit code 1
debian/rules:21: recipe for target 'override_dh_auto_configure' failed
make[1]: *** [override_dh_auto_configure] Error 255
make[1]: Leaving directory '/home/pi/knxd'
debian/rules:14: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2

How to solve this?

**Comment by smurfix** [comment]
You start with switching to the master branch.

    git checkout master
    git merge origin/master

as the "enable-ncn5120" option is clearly from 0.10. If that still doesn't work you post a few more lines of screen output, ideally to a pastebin or at least by clicking the "insert code" button here.

**Comment by gjniewenhuijse** [comment]
Thanks for your answer.

I'm on the master branch 0.12.1-3

See my complete output log on: http://pastebin.com/LQ4Exk8H

**Comment by smurfix** [comment]
> sudo dpkg-buildpackage -b -uc

Don't use sudo for building. You can undo this by running `sudo chown -R $USER .`

> configure: error: Unable to find a suitable libsystemd library

?? What's the output of
`dpkg -l libsystemd-dev  libsystemd-daemon-dev base-files`
?

**Comment by gjniewenhuijse** [comment]
I used for building (without sudo):  dpkg-buildpackage -b -uc > output.log 2>&1

```
pi@homekit ~/knxd $ dpkg -l libsystemd-dev libsystemd-daemon-dev base-files
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name           Version      Architecture Description
+++-==============-============-============-=================================
ii  base-files     7.1wheezy11+ armhf        Debian base system miscellaneous
dpkg-query: no packages found matching libsystemd-dev
dpkg-query: no packages found matching libsystemd-daemon-dev
```

**Comment by smurfix** [comment]
Commit 9627cd1 changed the default to requiring systemd. If you really need to build on Wheezy (personally I'd recommend switching to Jessie), you need to add `--without-systemd` to `debian/rules`

**Comment by gjniewenhuijse** [comment]
Sorry, i see the source in debian/rules file, but where in the file i need to add the --without-systemd ?

**Comment by smurfix** [comment]
Isn't that obvious?
the dh_auto_configure --all-the-.other-options line …

**Comment by timurapv** [comment]
Sorry, i'm newbe as well :(
Got the same error:
`configure: exit 1
dh_auto_configure: ./configure --build=arm-linux-gnueabihf --prefix=/usr --includedir=${prefix}/include --mandir=${prefix}/sh          are/man --infodir=${prefix}/share/info --sysconfdir=/etc --localstatedir=/var --disable-silent-rules --libdir=${prefix}/lib/a          rm-linux-gnueabihf --libexecdir=${prefix}/lib/arm-linux-gnueabihf --disable-maintainer-mode --disable-dependency-tracking --e          nable-usb --enable-groupcache --enable-ft12 --enable-ncn5120 --enable-tpuarts --enable-tpuarts-tcp --enable-dummy --libexecdi          r=/usr/lib --enable-dependency-tracking --enable-silent-rules returned exit code 1
debian/rules:21: recipe for target 'override_dh_auto_configure' failed
make[1]: *** [override_dh_auto_configure] Error 2
make[1]: Leaving directory '/home/pi/knxd'
debian/rules:14: recipe for target 'build' failed
make: *** [build] Error 2
dpkg-buildpackage: error: debian/rules build gave error exit status 2`
after _dpkg-buildpackage -d -b -uc_ command.

Help me please to install the **knxd**!

**Comment by smurfix** [comment]
It'd help if you (a) copied a few more lines, (b) used multi-line code tags (three backticks, not one).

**Comment by smurfix** [comment]
(c) copied with a working terminal program. In any case I do not see any  `--without-systemd` there, so please edit `debian/rules` correctly.

**Comment by timurapv** [comment]
Unfortunately i dont know how to use the multi-line code tags
So, here is my full log:
http://pastebin.com/W5TDCW9E

Thank you!

**Comment by smurfix** [comment]
Owch.

`dpkg-buildpackage -d` is nonsense. You've been asked for a missing dependency when you used `dpkg-buildpackage` without that option (or alternately, you've been following the installation instructions written by an idiot, but not me). Install `libev-dev` (there may be more) and try again. You might also need to remove the systemd stanzas from `debian/control`.

Also, the option to use is `--without-systemd`, not `--disable-`.

**Comment by timurapv** [comment]
> dpkg-buildpackage -d is nonsense. You've been asked for a missing dependency when you used dpkg-buildpackage without that option (or alternately, you've been following the installation instructions written by an idiot, but not me)

As i said, i new for a Linux...
How to understand folowing lines?
```
dpkg-buildpackage: warning: build dependencies/conflicts unsatisfied; aborting
dpkg-buildpackage: warning: (Use -d flag to override.)
```
That's why i used -d

**Comment by gjniewenhuijse** [comment]
I also didn't find a solution.

Added --without-systemd, but also an error while compiling, see:
http://pastebin.com/PGCexHnM

Sorry for using your limited time.

**Comment by smurfix** [comment]
conftest.cpp:36:2: error: #error "This is not a C++11 compiler"

Your system is too old. Use v0.10.

**Comment by mike-kfed** [comment]
this is just from the test if the compiler does c++11 without flags, he has gcc 4.9 and it's fine
`checking whether g++ supports C++11 features with -std=gnu++11... yes`
why he gets the `configure: error: Unable to find a suitable libsystemd library` at the end, I don't know, `--without-systemd` was specified, weird.

**Comment by mike-kfed** [comment]
I just checked out master and ran `./bootstrap.sh ; ./configure --without-systemd` and have the same error, this is a legitimate bug (or a need to document that knxd only works with systemd)

```
checking for systemd/sd-daemon.h... no
configure: error: Unable to find a suitable libsystemd library
```

**Comment by smurfix** [comment]
bah. Thanks for checking. Do you have time to track this down? if so, please prepare a pull request for oldstable (if the bug is reproducible there) or stable (if not), I'll forward merge it.

**Comment by mike-kfed** [comment]
hm, easy fix: correct use is `./configure --disable-systemd` maybe a change in m4 macros? I assume they have been updated or something.

**Comment by smurfix** [comment]
Meh. configure --help does reports the "--with-systemd" option, implying that one might be able to actually use --without-* to turn it off. Having both is confusing. As the --with-* option is used to specify the path if required, the --enable/--disable options should disappear / be aliased appropriately.

**Comment by smurfix** [comment]
fixed in commit 4a6cfd6
